#158792715 Set up a signup endpoint

elikeyz opened the pull request:
### What does this PR do?
Set up an endpoint for creating a new user account
#### Description of Task to be completed?
- Write a POST request for creating a new user account and generating a JSON Web Token
- Write tests for the endpoint
#### How should this be manually tested?
Test endpoint with Postman and run tests in the command line with "npm test"
#### What are the relevant pivotal tracker stories?
#158792715
#158800700

elikeyz committed:
feat(authentication): create API endpoint for signup

create an API endpoint to create a new user account
with a POST request
ensure that the response returns a json web token

[Delivers #158792715]

elikeyz committed:
chore(test): write tests for the signup endpoint

ensure that the tests pass

[Finishes #158800700]

fob413 commented:
I can see that your test coverage dropped a lot. Any plans on handling that?

fob413 commented on controllers/users.js:
```diff
@@ -0,0 +1,33 @@
+
+dotenv.config();
+
+const connectString = 'postgres://postgres:mastahacka@localhost:5432/rides';
```
Look into having this connection to the database done in a file somewhere and exported out. That way all you have to do in all your controller files is just to import the connected database.

elikeyz replied:
Ok

elikeyz committed:
fix(test): write more test cases

write more test cases for the sign up endpoint
and the validator middleware to improve test
coverage

[Fixes #158817814]

fob413 commented on utils/validate.js:
```diff
@@ -0,0 +1,15 @@
+  } else if (!req.body.email) {
+    res.status(400).send('Please enter your email');
+  } else if (!req.body.password) {
+    res.status(400).send('Please enter your password');
```
you've not tested this yet

elikeyz committed:
fix(test): write validator test for password field

ensure that the password field validator is tested

[Fixes #158817814]
